front/app/components/Form/Components/Controls/visibilityUtils.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ Reference for some of the code in this file: https://github.com/eclipsesource/jsonforms/blob/master/packages/core/src/testers/testers.ts
+ * 
+*/
+
+import Ajv from 'ajv';
+import {
+  UISchemaElement,
+  Rule,
+  Condition,
+  RuleEffect,
+  OrCondition,
+  AndCondition,
+  LeafCondition,
+  SchemaBasedCondition,
+  Scopable,
+  composeWithUi,
+  resolveData,
+} from '@jsonforms/core';
+import { has } from 'lodash-es';
+import { PageType } from '../Layouts/utils';
+
+export type ExtendedRule = {
+  /**
+   * The effect of the rule
+   */
+  effect: RuleEffect;
+  /**
+   * The condition of the rule that must evaluate to true in order
+   * to trigger the effect.
+   */
+  condition: Condition;
+} & Rule;
+
+export type ExtendedUISchema = {
+  ruleArray?: ExtendedRule[];
+  label?: string;
+} & UISchemaElement &
+  Scopable;
+
+const isOrCondition = (condition: Condition): condition is OrCondition =>
+  condition.type === 'OR';
+
+const isAndCondition = (condition: Condition): condition is AndCondition =>
+  condition.type === 'AND';
+
+const isLeafCondition = (condition: Condition): condition is LeafCondition =>
+  condition.type === 'LEAF';
+
+const isSchemaCondition = (
```

Comment:
Should we be supporting Or, And, and Leaf conditions? In the JSON Forms docs, it states that schema-based conditions are the new default :thinking: 
https://jsonforms.io/docs/uischema/rules/#ruleeffect